canonical-web-and-design/maas-ui
# fix(legacy): handle manger websocket errors with the error service

## Done

- Catch websocket errors from managers and pass them to the error service.

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Connect your UI to a local MAAS.
- Visit a legacy page.
- Stop your MAAS server.
- You should not see any websocket errors about isTrusted etc.
- Click on another legacy page, you should still not get any of those errors.

## Fixes

Fixes: canonical-web-and-design/maas-ui#1850.
Fixes: canonical-web-and-design/maas-ui#1874.
